### ISSUE TITLE
training time is slow because of PReLU

### ISSUE THREAD

**Issue by anti-machinee** [HabanaAI/Model-References#14]
- I train model use activation PReLU, 1 gaudi takes 60 hours for 1 epoch. It is decreased to 10 hours when I use ReLU instead. 
- Do you support PreLU already?
- If the reason is because Pytorch version?
Please help. thank you

**Comment by greg-serochi** [comment]
hi @anti-machinee, the SynapseAI Software stack does not yet support PReLU, so the activity here is falling back to run on CPU and taking more execution time.  Since we do support standard ReLU, this is why it's 6x faster when using ReLU instead.

The team did a check on some options here, moving your OPS to ReLU would be ideal, but there are other options to try, inspired by [this](https://stackoverflow.com/questions/39975676/how-to-implement-prelu-activation-in-tensorflow): prelu(x) = relu(x) + alpha*(x - abs(x)) * 0.5

By the way, you can see the PyTorch OPS support in our documentations [here](https://docs.habana.ai/en/latest/PyTorch/Pytorch_Operators/Pytorch_Operators.html)

**Comment by anti-machinee** [comment]
@greg-serochi PReLU in Pytorch has an alpha which is learnable parameter. If I use only PreLU as formulation PReLU(x)=max(0,x)+a∗min(0,x) the speed is as same as ReLU but I just want to use PReLU with trainable parameter. Can you explain 
1. Why PReLU runs slowly in SynapseAI Software but fast in CUDA
2. Are there any trick to use a learnable parameters in PReLu as default Pytorch

**Comment by anti-machinee** [comment]
Thank you @ssarkar2 
I write PReLU by myself and it works, speed decrease from 7 hours (8 gaudi) to 2.5 hours (8 gaudi) and model is different from the model use PReLU default but accuracy does not change much
Another one, I use Pytorch PReLU formulation and this one is different from the formulation you use.

class PReLUCustom(nn.Module):

    def __init__(self, num_parameters):
        super(PReLUCustom, self).__init__()
        self.weight = nn.Parameter()
        self.weight = nn.Parameter(torch.FloatTensor(num_parameters))  # out, in
        # nn.init.xavier_uniform_(self.weight)
        self.relu = nn.ReLU()
    def forward(self, _input):
        zero_tmp = torch.zeros_like(_input)
        x = self.relu(_input) + self.weight.unsqueeze(1).unsqueeze(2).expand_as(_input) * torch.min(zero_tmp, _input)
        return x

**Comment by anti-machinee** [comment]
Would PReLU be supported soon ?
